src/EventListener/ProductPublisher.php

```diff
@@ -38,8 +60,27 @@ public function onFlush(OnFlushEventArgs $event)
         $scheduledInsertions = $event->getEntityManager()->getUnitOfWork()->getScheduledEntityInsertions();
 
         foreach ($scheduledInsertions as $entity) {
-            if ($entity instanceof ProductInterface) {
-                $this->scheduledProducts[] = $entity;
+            $entity = $this->getProductFromEntity($entity);
+            if ($entity instanceof ProductInterface && !isset($this->scheduledProducts[$entity->getId()])) {
+                $this->scheduledProducts[$entity->getId()] = $entity;
+            }
+        }
+
+        $scheduledUpdates = $event->getEntityManager()->getUnitOfWork()->getScheduledEntityUpdates();
+
+        foreach ($scheduledUpdates as $entity) {
+            $entity = $this->getProductFromEntity($entity);
+            if ($entity instanceof ProductInterface && !isset($this->scheduledUpdates[$entity->getId()])) {
+                $this->scheduledUpdates[$entity->getId()] = $entity;
+            }
+        }
+
+        $scheduledDeletions = $event->getEntityManager()->getUnitOfWork()->getScheduledEntityDeletions();
+
+        foreach ($scheduledDeletions as $entity) {
+            /** We delete only if the product itself was removed */
```

Comment:
I think we also need the case we an related image is deleted, then the product should be updated

Reply:
Makes sense